File: hw9mvp/src/main/java/com/homework/hw8mvp/SettingsActivity.java
package com.homework.hw8mvp;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.google.android.material.switchmaterial.SwitchMaterial;

public class SettingsActivity extends AppCompatActivity {

    private SwitchMaterial aSwitch;
    private boolean temperatureCheck;
    private SharedPreferences sPref;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_settings);

        aSwitch = findViewById(R.id.switchTemperature);

        boolean b = loadBoolean();
        aSwitch.setChecked(loadBoolean());
        aSwitch.setOnCheckedChangeListener((buttonView, isChecked) -> {
            if (isChecked) {
                temperatureCheck = true;
                Toast.makeText(SettingsActivity.this, "Fahrenheit mode is on", Toast.LENGTH_SHORT).show();
                saveBoolean(temperatureCheck);
            } else {
                temperatureCheck = false;
                Toast.makeText(SettingsActivity.this, "Included Celsius mode", Toast.LENGTH_SHORT).show();
                saveBoolean(temperatureCheck);
            }
        });

        findViewById(R.id.backButton).setOnClickListener(v -> {
            Intent intent = new Intent(SettingsActivity.this, MainActivity.class);
            intent.putExtra("TEMPCHECK", temperatureCheck);
            startActivity(intent);
            finish();
        });
    }

    private void saveBoolean(boolean temperatureCheck) {
        sPref = getPreferences(MODE_PRIVATE);
        SharedPreferences.Editor ed = sPref.edit();
        ed.putBoolean("ISCHEKED", temperatureCheck);
        ed.apply();
    }

    private boolean loadBoolean() {
        sPref = getPreferences(MODE_PRIVATE);
        this.temperatureCheck = sPref.getBoolean("ISCHEKED", false);
        return temperatureCheck;
    }

}

File: hw5_2/src/main/java/com/homework/hw5_2/SettingsActivity.kt
package com.homework.hw5_2

import android.content.SharedPreferences
import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity


class SettingsActivity : AppCompatActivity() {
    private lateinit var sPref: SharedPreferences
    private var task = 1

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_settings)

        findViewById<View>(R.id.radioButton1).setOnClickListener {
            task = 1
            saveTask(task)
            Toast.makeText(this@SettingsActivity,
                    "Mode \"ThreadPoolExecutor and Handler\"", Toast.LENGTH_SHORT).show()
            finish()
        }
        findViewById<View>(R.id.radioButton2).setOnClickListener {
            task = 2
            saveTask(task)
            Toast.makeText(this@SettingsActivity,
                    "Mode \"CompletableFuture and ThreadPoolExecutor\"", Toast.LENGTH_SHORT).show()
            finish()
        }
        findViewById<View>(R.id.radioButton3).setOnClickListener {
            task = 3
            saveTask(task)
            Toast.makeText(this@SettingsActivity,
                    "Mode \"RxJava\"", Toast.LENGTH_SHORT).show()
            finish()
        }
    }

    private fun saveTask(task: Int) {
        sPref = getSharedPreferences("SettingsActivity", MODE_PRIVATE)
        val ed = sPref.edit()
        ed.putInt("TASK", task)
        ed.apply()
    }
}